packages/react-reconciler/src/ReactFiberHooks.js

```diff
@@ -292,6 +294,13 @@ function areHookInputsEqual(
   nextDeps: Array<mixed>,
   prevDeps: Array<mixed> | null,
 ) {
+  if (__DEV__) {
+    // Force to remount effects, callbacks, and memo for hot reloaded components.
+    if (isInvalidatedForHotReloading(currentlyRenderingFiber)) {
```

Comment:
Why is this in this function? I would expect it to be in `renderWithHooks`, since that is called once per component instead of once per hook with deps.

Reply:
I want to reset every Hook with dependencies, but keep state of other Hooks. So `useEffect`, `useCallback`, `useMemo`, `useImperativeHandle` should behave as if their deps changed (even if they were `[]` in the first place). But `useRef`, `useState`, `useReducer` should retain their values.

I wasn't sure what's an easy way to write this in `renderWitHooks`. I'd have to traverse the list and tear them down manually, but they don't have "types" in the list so I'd need to add those?

It seemed easier to just override the comparison to force any dep comparison to fail.

Reply:
I was thinking you would set a module variable called something like `ignorePreviousDependencies = isInvalidatedForHotReloading(fiber)` inside  `renderWithHooks`, and then use that in `areHookInputsEqual`.

I suppose it's not a big deal since it's DEV but comparing deps is a fairly hot code path, and this adds an unconditional map lookup.

Reply:
I generally don't like adding new module level variables but since this one would only be set at the beginning of `renderWithHooks`, and remain the same throughout the entire phase, seems ok.

Reply:
Ah I see what you mean now. Thanks for explaining. This makes sense.

Reply:
Pushed.